src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2887 @@
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Declare variables.
+!-----------------------------------------------------------------------
+
+  REAL(r_kind),INTENT(IN   ) :: N0,alpha,alp_a,alp_b,beta_a,beta_b,lamda
+
+  REAL(r_kind),INTENT(  OUT) :: refRainHV
+
+  !local variables
+  REAL(r_kind) :: expon_hv
+  REAL(r_kind) :: gamma_hv
+  REAL(r_kind) :: N0_units
+
+  REAL(r_double) :: gamma
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+  N0_units = (1.e-3_r_kind)**(4.0_r_kind+alpha)
+
+  gamma_hv = sngl(gamma(dble(beta_a)+dble(beta_b)+dble(alpha)+1.0_r_double))
+  expon_hv = -(alpha+beta_a+beta_b+1)
+
+  refRainHV = mm3todBZ*radar_const*alp_a*alp_b*(N0*N0_units)*           &
+                gamma_hv*(lamda*1.e-3_r_kind)**expon_hv
+
+END SUBROUTINE partialRhoRain
+
+
+SUBROUTINE partialRefIce(N0,alpha,Ai,Bi,Ci,alp_a,alp_b,beta_a,beta_b,   &
+                         lamda,refIceHH,refIceVV)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! This function calculates the partial reflectivity for each species
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 1/22/2007
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Declare variables.
+!-----------------------------------------------------------------------
+  REAL(r_kind),INTENT(IN   ) :: N0,alpha,Ai,Bi,Ci,alp_a,alp_b,beta_a,beta_b
+  REAL(r_kind),INTENT(IN   ) :: lamda
+
+  REAL(r_kind),INTENT(  OUT) :: refIceHH,refIceVV
+
+  !local variables
+  REAL(r_kind) :: gamma_h, gamma_v, expon_h, expon_v
+  REAL(r_kind) :: N0_units
+  REAL(r_double) :: gamma
+
+  gamma_h = sngl(gamma(dble(alpha) + 2.0_r_double*dble(beta_a)+1.0_r_double))
+  expon_h = -(alpha+2*beta_a+1)
+  gamma_v = sngl(gamma(dble(alpha) + 2.0_r_double*dble(beta_b)+1.0_r_double))
+  expon_v = -(alpha + 2*beta_b+1)
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+  N0_units = (1.e-3_r_kind)**(4.0_r_kind+alpha)
+
+  refIceHH = mm3toDBZ*radar_Const*gamma_h*(N0*N0_units)*                     &
+              (Ai*alp_a**2+Bi*alp_b**2+2*Ci*alp_a*alp_b)*                 &
+             (lamda*1.e-3_r_kind)**expon_h
+
+  refIceVV = mm3toDBZ*radar_Const*gamma_v*(N0*N0_units)*                    &
+             (Bi*alp_a**2+Ai*alp_b**2+2*Ci*alp_a*alp_b)*                  &
+             (lamda*1.e-3_r_kind)**expon_v
```

Comment:
Please add _r_kind to 2
```
(Ai*alp_a**2+Bi*alp_b**2+2_r_kind*Ci*alp_a*alp_b) - line 1658
(Bi*alp_a**2+Ai*alp_b**2+2_r_kind*Ci*alp_a*alp_b) - line 1662
```

Reply:
We changed the code as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R1657-R1663